GameLogic/python_emulator.py

```diff
@@ -0,0 +1,531 @@
+        )
+        self.cactus_array = np.array(
+            [
+                [0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 0, 0],
+                [0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0],
+                [0, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0],
+                [0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+            ]
+        )
+        self.mid_bird_array = np.array(
+            [
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+                [0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0],
+            ]
+        )
+
+    def setup_parameters(self):
+
+        # Converts user ratios for the pygame GUI display
+        self.JUMP_HEIGHT = self.U_JUMP_HEIGHT * self.GRID_SIZE
+        self.WIDTH, self.HEIGHT = (
+            self.U_WIDTH * self.GRID_SIZE,
+            self.U_HEIGHT * self.GRID_SIZE,
+        )
+
+        # Creates reference axis for objects to move along
+        self.GND_AXIS = self.HEIGHT - self.GND * self.GRID_SIZE  # ground line in black
+        self.REF_AXIS = (
+            self.HEIGHT - self.REF * self.GRID_SIZE
+        )  # main line the dino and cactus are on
+        self.MID_BIRD_AXIS = (
+            self.HEIGHT - self.BIRD * self.GRID_SIZE
+        )  # mid bird flying line
+
+        # Block shape and size
+        self.dino_array = self.jump_array
+        self.dino_height, self.dino_width = self.dino_array.shape
+        self.dino_x, self.dino_y = 10, self.REF_AXIS - self.dino_height * self.GRID_SIZE
+        self.original_dino_y = self.dino_y
+        self.cactus_height, self.cactus_width = self.cactus_array.shape
+        self.cactus_y = self.REF_AXIS - self.cactus_height * self.GRID_SIZE
+        self.mid_bird_height, self.mid_bird_width = self.mid_bird_array.shape
+        self.mid_bird_y = self.MID_BIRD_AXIS - self.mid_bird_height * self.GRID_SIZE
+        self.curr_walk = self.UPDATE_WALK_FRAMES
```

Comment:
`dino_x`, all the `height` and `width`, `cactus_y` and `mid_bird_y` can be considered consts?

Reply:
and instead of putting `dino_x` as 10, should it be based on `GRID_SIZE`?